File: DistributedWebCrawler.Core/Models/IngestFailureReason.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DistributedWebCrawler.Core.Model
{
    public enum IngestFailureReason
    {
        None = 0,
        UnknownError,
        MaxDepthReached,
        Http4xxError,
        NetworkConnectivityError,
        UriFormatError,
        RequestTimeout,
        ContentTooLarge,
        MediaTypeNotPermitted,
        MaxRedirectsReached
    }
}

File: DistributedWebCrawler.ManagerAPI/src/ts/types/IngestFailureReason.ts

export const enum IngestFailureReason {
    None = 0,
    UnknownError = 1,
    MaxDepthReached = 2,
    Http4xxError = 3,
    NetworkConnectivityError = 4,
    UriFormatError = 5,
    RequestTimeout = 6,
    ContentTooLarge = 7,
    MediaTypeNotPermitted = 8,
    MaxRedirectsReached = 9
}